web/src/stream_settings_ui.js

```diff
@@ -872,14 +872,27 @@ export function toggle_view(event) {
     const active_data = stream_settings_components.get_active_data();
     const stream_filter_tab = active_data.$tabs.first().text();
 
-    if (event === "right_arrow" && stream_filter_tab === "Subscribed") {
-        toggler.goto("not-subscribed");
-    } else if (event === "right_arrow" && stream_filter_tab === "Not subscribed") {
-        toggler.goto("all-streams");
-    } else if (event === "left_arrow" && stream_filter_tab === "Not subscribed") {
-        toggler.goto("subscribed");
-    } else if (event === "left_arrow" && stream_filter_tab === "All channels") {
-        toggler.goto("not-subscribed");
+    switch (event) {
+        case "right_arrow":
+            switch (stream_filter_tab) {
+                case "Subscribed":
+                    toggler.goto("not-subscribed");
+                    break;
+                case "Not subscribed":
+                    toggler.goto("all-streams");
+                    break;
+            }
+            break;
+        case "left_arrow":
+            switch (stream_filter_tab) {
+                case "Not subscribed":
+                    toggler.goto("subscribed");
+                    break;
+                case "All channels":
```

Comment:
I'm skeptical of this logic keying off what looks like a translated string; I think this code might not work for non-English user languages.

We should be able to use a data attribute for these conditionals, rather than reading text from the DOM.

This isn't a new issue in this PR, though; can you just fix that in a quick follow-up PR @sujalshah-bit?